symfit/core/fit.py

```diff
@@ -488,148 +662,85 @@ def jacobian(self):
             Variable's. The return shape is a list over the models components,
             filled with tha symbolical jacobian for that component, as a list.
         """
-        return [[sympy.diff(expr, param) for param in self.params] for expr in self.values()]
-
-    @property
-    def hessian(self):
-        """
-        :return: Hessian filled with the symbolic expressions for all the
-            second order partial derivatives. Partial derivatives are taken with
-            respect to the Parameter's, not the independent Variable's.
-        """
-        return [[[sympy.diff(partial_dv, param) for param in self.params]
-                 for partial_dv in comp] for comp in self.jacobian]
-
-    @property
-    def chi_squared(self):
-        """
-        :return: Symbolic :math:`\\chi^2`
-        """
-        return sum(((f - y)/self.sigmas[y])**2 for y, f in self.items())
-
-    @property
-    def chi(self):
-        """
-        :return: Symbolic Square root of :math:`\\chi^2`. Required for MINPACK
-            optimization only. Denoted as :math:`\\sqrt(\\chi^2)`
-        """
-        return sympy.sqrt(self.chi_squared)
-
-    @property
-    def chi_jacobian(self):
-        """
-        Return a symbolic jacobian of the :math:`\\sqrt(\\chi^2)` function.
-        Vector of derivatives w.r.t. each parameter. Not a Matrix but a vector!
-        This is because that's what leastsq needs.
-        """
         jac = []
-        for param in self.params:
-            # Differentiate to every param
-            f = sympy.diff(self.chi, param)
-            jac.append(f)
+        for var, expr in self.items():
+            jac.append([])
+            for param in self.params:
+                partial_dv = D(var, param)
+                jac[-1].append(self.jacobian_model[partial_dv])
```

Comment:
I would build the new list first, and append it at the end instead of vice-versa